ui/src/map/layers/utils/switch-layer-utils.ts

```diff
@@ -14,33 +20,37 @@ import { findMatchingEntities, pointToCoords } from 'map/layers/utils/layer-util
 import { SearchItemsOptions } from 'map/layers/utils/layer-model';
 import VectorSource from 'ol/source/Vector';
 import { Rectangle } from 'model/geometry';
+import { ValidatedAsset } from 'publication/publication-model';
 
 const switchImage: HTMLImageElement = new Image();
 switchImage.src = `data:image/svg+xml;utf8,${encodeURIComponent(SwitchIcon)}`;
 
+const switchErrorImage: HTMLImageElement = new Image();
+switchErrorImage.src = `data:image/svg+xml;utf8,${encodeURIComponent(SwitchErrorIcon)}`;
```

Comment:
Uusi ikoni virheikkäälle vaihteelle. Varmaan tämänkin voisi jotenkin tyylikikkailla tuosta alkuperäisestä ikonista, mutta mun SVG-fu loppuu sen kanssa kesken.

Reply:
Olikos tuossa muuta kuin värivaihto? En siis itsekään tiedä suorilta miten tuo canvas-piirrossa tehtäisiin, mutta voisi olla kyllä ihan hyödyllinen juttu tutkia/funktioida jatkoakin ajatellen ettei värien takia tarvitse monistaa. Ei toki mitenkään must tässä yhteydessä.

Reply:
Karin alkuperäisessä versiossa tästä branchista tuo oli tehtynä värinvaihdolla. Se oli vaan aika häkki (merkittykin TODO-kommentilla "hack to be solved",) näytti tältä

```
switchErrorImage.src = `data:image/svg+xml;utf8,${encodeURIComponent(SwitchIcon.replace('black', styles.errorDefault))}`;
```

Kyllähän tuokin toimii, mutta on mielestäni aika rumaa. Muuten tässä voisi käyttää vaan suoraan CSS-tyylejä niinkuin me tehdään about kaikkialla muualla ikoneiden kanssa, mutta en ole ihan varma kuinka hyvin se toimisi näiden karttacanvakselle rendattujen ikoneiden kanssa. Täytyisi tutkia asiaa